.config/doom/config.org
#+title: My Config
#+PROPERTY: header-args :tangle config.el
#+STARTUP: overview
#+auto_tangle: t

* Useful config
** name
#+begin_src emacs-lisp
(setq user-full-name "Wim Veto"
      user-mail-address "wimveto@gmail.com")
#+end_src
** font
#+begin_src emacs-lisp
(setq doom-font (font-spec :family "JetBrainsMono NF" :size 15))
#+end_src
** theme
#+begin_src emacs-lisp
(setq doom-theme 'doom-moonlight)
#+end_src
** relative line numbers
#+begin_src emacs-lisp
(setq display-line-numbers-type 'relative)
#+end_src
** transparency
#+begin_src emacs-lisp
(add-to-list 'default-frame-alist '(alpha-background . 95))
#+end_src
** bookmark file
#+begin_src emacs-lisp
(setq bookmark-default-file "~/.config/doom/bookmarks")
#+end_src
** projectile directory
#+begin_src emacs-lisp
(setq projectile-project-search-path '("~/Code/web/"))
#+end_src
** doom modeline
#+begin_src emacs-lisp
(setq doom-modeline-bar-width 5)
;; (display-battery-mode 1)
;; (setq doom-modeline-battery t)
;; (display-time-mode 1)
;; (setq display-time-format "%Y %b %d (%a) %R %p")
;; (setq doom-modeline-time t)
(setq doom-modeline-height 30)
#+end_src
** emacsclient
#+begin_src emacs-lisp
(after! persp-mode
  (setq persp-emacsclient-init-frame-behaviour-override "main"))
#+end_src
** custom bindings
#+begin_src emacs-lisp
(map! :leader
      :prefix ("w" . "window")
      :desc "writeroom mode" "M-w" #'writeroom-mode)
#+end_src
* Org
** org roam
*** default directory
#+begin_src emacs-lisp
(setq org-roam-directory "~/Org/roam/mindmap")
#+end_src
*** custom functions
**** switch org-roam directory
#+begin_src emacs-lisp
(setq zettlekasten-paths-alist '(("Main" . "~/Org/roam/mindmap")
                                 ("Noosphere" . "~/Org/roam/stories/noosphere")))

(defun my/switch-zettlekasten ()
  (interactive)
  (let* ((keys (mapcar #'car zettlekasten-paths-alist))
         (prompt (format "Select Zettlekasten:"))
         (key (completing-read prompt keys))
         (chosen-zettlekasten-path (cdr (assoc key zettlekasten-paths-alist))))
    (setq org-roam-directory chosen-zettlekasten-path)
    (setq org-roam-db-location (file-name-concat chosen-zettlekasten-path "org-roam.db"))
    (org-roam-db-sync)))
#+end_src
**** print org-roam directory
#+begin_src emacs-lisp
(defun my/print-org-roam-directory ()
  "Print the value of org-roam-directory to the minibuffer."
  (interactive)
  (message "org-roam-directory: %s" org-roam-directory))
#+end_src
*** custom bindings
#+begin_src emacs-lisp
(map! :leader
    :prefix ("n" . "notes")
    (:prefix ("r" . "roam")
     :desc "Switch to an Org Roam database" "S" #'my/switch-zettlekasten
     :desc "Open Org Roam UI" "o" #'org-roam-ui-open
     :desc "Print current Org Roam directory" "v" #'my/print-org-roam-directory))
#+end_src
** org roam ui
#+begin_src emacs-lisp
(use-package! websocket
    :after org-roam)
(use-package! org-roam-ui
    :after org-roam ;; or :after org
;;         normally we'd recommend hooking orui after org-roam, but since org-roam does not have
;;         a hookable mode anymore, you're advised to pick something yourself
;;         if you don't care about startup time, use
;;  :hook (after-init . org-roam-ui-mode)
    :config
    (setq org-roam-ui-sync-theme t
          org-roam-ui-follow t
          org-roam-ui-update-on-save t
          org-roam-ui-open-on-start t))
#+end_src
** org agenda
#+begin_src emacs-lisp
(after! org
  (setq org-agenda-files
        '("~/Org/agenda/agenda.org")))
#+end_src

** org auto-tangle
#+begin_src emacs-lisp
(use-package! org-auto-tangle
  :defer t
  :hook (org-mode . org-auto-tangle-mode)
  :config
  (setq org-auto-tangle-default t))
#+end_src
* Evil
** custom bindings
#+begin_src emacs-lisp
(use-package! evil-org
  :config
  (map! :map evil-org-mode-map
        :i "C-k" #'evil-insert-digraph))
#+end_src
* Programming
** highlight indents
#+begin_src emacs-lisp
(setq highlight-indent-guides-method 'column)
(setq highlight-indent-guides-responsive 'top)
#+end_src
** web mode
#+begin_src emacs-lisp
(map! :after web-mode
      :map emmet-mode-keymap
      :i
      "<tab>" #'emmet-expand-line)

(setq web-mode-enable-current-element-highlight t)
#+end_src


.config/doom/config.el
(setq user-full-name "Wim Veto"
      user-mail-address "wimveto@gmail.com")

(setq doom-font (font-spec :family "JetBrainsMono NF" :size 15))

(setq doom-theme 'doom-moonlight)

(setq display-line-numbers-type 'relative)

(add-to-list 'default-frame-alist '(alpha-background . 95))

(setq bookmark-default-file "~/.config/doom/bookmarks")

(setq projectile-project-search-path '("~/Code/web/"))

(setq doom-modeline-bar-width 5)
;; (display-battery-mode 1)
;; (setq doom-modeline-battery t)
;; (display-time-mode 1)
;; (setq display-time-format "%Y %b %d (%a) %R %p")
;; (setq doom-modeline-time t)
(setq doom-modeline-height 30)

(after! persp-mode
  (setq persp-emacsclient-init-frame-behaviour-override "main"))

(map! :leader
      :prefix ("w" . "window")
      :desc "writeroom mode" "M-w" #'writeroom-mode)

(setq org-roam-directory "~/Org/roam/mindmap")

(setq zettlekasten-paths-alist '(("Main" . "~/Org/roam/mindmap")
                                 ("Noosphere" . "~/Org/roam/stories/noosphere")))

(defun my/switch-zettlekasten ()
  (interactive)
  (let* ((keys (mapcar #'car zettlekasten-paths-alist))
         (prompt (format "Select Zettlekasten:"))
         (key (completing-read prompt keys))
         (chosen-zettlekasten-path (cdr (assoc key zettlekasten-paths-alist))))
    (setq org-roam-directory chosen-zettlekasten-path)
    (setq org-roam-db-location (file-name-concat chosen-zettlekasten-path "org-roam.db"))
    (org-roam-db-sync)))

(defun my/print-org-roam-directory ()
  "Print the value of org-roam-directory to the minibuffer."
  (interactive)
  (message "org-roam-directory: %s" org-roam-directory))

(map! :leader
    :prefix ("n" . "notes")
    (:prefix ("r" . "roam")
     :desc "Switch to an Org Roam database" "S" #'my/switch-zettlekasten
     :desc "Open Org Roam UI" "o" #'org-roam-ui-open
     :desc "Print current Org Roam directory" "v" #'my/print-org-roam-directory))

(use-package! websocket
    :after org-roam)
(use-package! org-roam-ui
    :after org-roam ;; or :after org
;;         normally we'd recommend hooking orui after org-roam, but since org-roam does not have
;;         a hookable mode anymore, you're advised to pick something yourself
;;         if you don't care about startup time, use
;;  :hook (after-init . org-roam-ui-mode)
    :config
    (setq org-roam-ui-sync-theme t
          org-roam-ui-follow t
          org-roam-ui-update-on-save t
          org-roam-ui-open-on-start t))

(after! org
  (setq org-agenda-files
        '("~/Org/agenda/agenda.org")))

(use-package! org-auto-tangle
  :defer t
  :hook (org-mode . org-auto-tangle-mode)
  :config
  (setq org-auto-tangle-default t))

(use-package! evil-org
  :config
  (map! :map evil-org-mode-map
        :i "C-k" #'evil-insert-digraph))

(setq highlight-indent-guides-method 'column)
(setq highlight-indent-guides-responsive 'top)

(map! :after web-mode
      :map emmet-mode-keymap
      :i
      "<tab>" #'emmet-expand-line)

(setq web-mode-enable-current-element-highlight t)
